### PR TITLE
Improve handling of multiple subevents and particle extraction

### LINKED ISSUE #13: Issue with `from icecube.icetray import I3Tray` in container
When running against a modern IceTray installation one gets a warning about `from I3Tray import I3Tray` [being deprecated](https://docs.icecube.aq/icetray/main/projects/icetray/I3Tray.html).

However in CI `from icecube.icetray import I3Tray` fails due to `I3Tray` not having an `Add` attribute.

Currently I had to revert PR #7 to the old-style import for the sake of CI testing.

### LINKED ISSUE #14: `i3_to_json.py` issue with multiple subevents file
It seems the current version of the convert script has troubles processing `/data/ana/realtime/alert_catalog_v2/input_files/Level2pass2_IC86.2013_data_Run00124340_Subrun00000000_00000052_event19400842.i3.zst`

The first event is written out, but the second event fails when retrieving the keys from the dictionary: 
```
File "[...]/skywriter/i3_to_json.py", line 97, in fill_missing_keys
    pframe = source_pframes[uid]
KeyError: (124340, 19400842, 1)
```

To be debugged.

### PR DESCRIPTION
**Pull request by mlincett**
This PR fixes #14 with a more graceful handling of multiple subevents produced by `I3TriggerSplitter`.

The core changes are:
1. the relevant I3 modules are now run only for subevents where the `uid` matches the one in the original P-frame;
2. almost no logic is run on P frames that are meant to be skipped, hence the exception handling has been largely simplified;
3. the custom extractor for "extra particles" is removed: extra keys are retrieved from the original P-frame (as suggested by @tianluyuan before).

Bonus changes:
- fixes #13 by updating IceTray base image;
- the log formatting is improved, printing log level as well as the pending particle ID;
- log config and main logger defined in a `logging_utils` module that replaces/incorporates the previous `suppress_warnings` module/logic;
- a test file with multiple subevents is added;
- testing logic is extended with the I3Particle extraction functionality (`--extra`) .

### CODE REVIEW

File: skywriter/i3_to_json.py [hunk 1]
```diff
@@ -383,7 +383,7 @@ def main():
     parser.add_argument("-o", "--out", default="", help="output I3 file")
     args = parser.parse_args()
 
-    logging_tools.log_argparse_args(args)
+    logging_tools.log_argparse_args(args, logger=LOGGER, level="DEBUG")
```
Comment (ric-evans) [inline skywriter/i3_to_json.py:386]: It could be nice to make the log-level user configurable

**Review by ric-evans — CHANGES_REQUESTED** [review]

**Comment by mlincett** [comment]
> This looks good. Just to clarify is the idea that I3Particles from the original p-frame would still be user specified via `--extra`, including those needed in the online filtering? Or are some keys in the original p-frame always saved by default? Maybe a README would be nice to add to explain the behavior.

All the keys required for the realtime processing are copied from the input to the output P-frame.

The coordinates of I3Particles specified in the `--extra` keys are extracted and written in plaintext to the JSON, but there is no warranty that the corresponding data will be in the output I3Frame (as the user could choose a key not among the ones that are routinely copied). This is no different from the current `main` branch behavior, I believe (but of course it can be improved in the future).

I agree this can be clarified in a README.

**Review by tianluyuan — APPROVED** [review]

File: skywriter/logging_utils.py [hunk 2]
```diff
@@ -1,6 +1,15 @@
-# temporary workaround for https://github.com/icecube/icetray/issues/3112
 import warnings
+import logging
 
+# temporary workaround for https://github.com/icecube/icetray/issues/3112
 warnings.filterwarnings(
     "ignore", ".*already registered; second conversion method ignored.", RuntimeWarning
 )
+
+logging.basicConfig(
```
[the remark this answers is not in the document]
Comment (mlincett) [reply skywriter/logging_utils.py]: My purpose is configure the logging for `i3_to_json.py` when run as script, so I guess the solution is putting the `logging.basicConfig` before the call to `main()`. If you have a better suggestion please let me know!

**Comment by ric-evans** [comment]
> > Looking forward, we'll need to define an entry point pattern. I see two equally suitable patterns:
> > ```
> > 1. multiple entry points: each module has an `if-main` block
> >    
> >    * multiple `argparse`s
> > 
> > 2. one entry point: make a `__main__.py`
> >    
> >    * one `argparse`
> > ```
> 
> We don't really have actual multiple modules now, do we?
> 
> I mostly use the `i3_to_json.py` as a script for testing but then I use `i3_to_json` as a function in SkyMist for all practical purposes. So no strong preference in either direction.

You're right, we don't right now. However, if you wanted to go with one entry point, then there'd need to be refactoring that would be easier to do now than later when we add a new script. BUT, it sounds like multiple entry points are fine :smile: we'll just have to use `python -m` so the helper module is available

**Review by ric-evans — APPROVED** [review]

**Pull request merged**